app-modules/interaction/src/Filament/Actions/DraftWithInteractionAiAction.php

```diff
@@ -0,0 +1,161 @@
+use AdvisingApp\Ai\Settings\AiSettings;
+use Filament\Forms\Components\Textarea;
+use Filament\Notifications\Notification;
+use Filament\Forms\Components\Actions\Action;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\Interaction\Models\InteractionType;
+use AdvisingApp\Interaction\Models\InteractionDriver;
+use AdvisingApp\Interaction\Models\InteractionOutcome;
+use AdvisingApp\Ai\Exceptions\MessageResponseException;
+use AdvisingApp\Interaction\Models\InteractionInitiative;
+use AdvisingApp\Ai\Settings\AiIntegratedAssistantSettings;
+
+class DraftWithInteractionAiAction extends Action
+{
+    protected array | Closure $mergeTags = [];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this
+            ->label('Draft with AI Assistant')
+            ->link()
+            ->icon('heroicon-m-pencil')
+            ->modalContent(fn (Page $livewire) => view('interaction::filament.actions.draft-with-ai-modal-content', [
+                'recordTitle' => $livewire->record['full_name'],
+                'avatarUrl' => AiAssistant::query()->where('is_default', true)->first()
+                    ?->getFirstTemporaryUrl(now()->addHour(), 'avatar', 'avatar-height-250px') ?: Vite::asset('resources/images/canyon-ai-headshot.jpg'),
+            ]))
+            ->modalWidth(MaxWidth::ExtraLarge)
+            ->modalSubmitActionLabel('Draft')
+            ->form([
+                Textarea::make('instructions')
+                    ->hiddenLabel()
+                    ->rows(4)
+                    ->placeholder('What do you want to write about?')
+                    ->required(),
+            ])
+            ->action(function (array $data, Get $get, Set $set, Page $livewire) {
+                $service = Feature::active('ai-integrated-assistant-settings')
+                    ? app(AiIntegratedAssistantSettings::class)->default_model->getService()
+                    : app(AiSettings::class)->default_model->getService();
+
+                $userName = auth()->user()->name;
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $modal = $livewire::getResource()::getPluralModelLabel();
+                $initiative = InteractionInitiative::find($get('interaction_initiative_id'));
+                $driver = InteractionDriver::find($get('interaction_driver_id'));
+                $outcome = InteractionOutcome::find($get('interaction_outcome_id'));
+                $type = InteractionType::find($get('interaction_type_id'));
+
+                try {
+                    $content = $service->complete(
+                        <<<EOL
+                            My name is {$userName}, and I am a {$userJobTitle} at {$clientName}.
+
+                            Please document my interaction with {$modal} at our college based on the following details:
```

Comment:
Right now this would translate down to `Please document my interaction with Student (or Prospects) at our college based on the following details`.
Which doesn't make a lot of sense.

Let's change it to something like:
`Please document my interaction with the {$model} $livewire->record['full_name'] at our college based on the following details`
and I would change `$modal` to get the model label, not the plural label.

This would translate down to:
`Please document my interaction with the Student (or Prospect) Jane Doe at our college based on the following details`